lib/presentation/profile/bloc/profile_bloc.dart

```diff
@@ -0,0 +1,171 @@
+
+      _activityDetails = await UserRepository.getActivityDetails(_user!.id!);
+    } on Exception catch (_) {
+      emit(state.copyWith(status: const Status.error(AppStrings.genericError)));
+      return;
+    }
+
+    for (final activity in _activityDetails ?? <ActivityDetails>[]) {
+      if (activity.createdAt == null) continue;
+      _activity[activity.createdAt!] = activity.correct;
+    }
+
+    bool? _isFollowing;
+    for (final follower in _followingList ?? <Following>[]) {
+      if (follower.id == event.userId) {
+        _isFollowing = true;
+        break;
+      }
+    }
+
+    _isFollowing ??= false;
+
+    emit(state.copyWith(
+      status: const Status(),
+      user: _user,
+      following: _followingList,
+      submissionStatus: _subStats,
+      personalProfile: event.userId.isEmpty,
+      isFollowing: _isFollowing,
+      currentYear: _currentYear,
+      currentTriplet: _currentTriplet,
+      showFollowing: false,
+    ));
+  }
+
+  void _onUpdateYear(UpdateYear event, Emitter<ProfileState> emit) {
+    if (event.increment) {
+      _currentYear++;
+    } else {
+      _currentYear--;
+    }
+
+    emit(state.copyWith(currentYear: _currentYear));
+  }
+
+  void _onUpdateMonth(UpdateMonth event, Emitter<ProfileState> emit) {
+    if (event.increment) {
+      if (_currentTriplet == 3) _currentYear++;
+      _currentTriplet++;
+    } else {
+      if (_currentTriplet == 0) _currentYear--;
+      _currentTriplet--;
+    }
+
+    _currentTriplet %= 4;
+
+    emit(state.copyWith(
+      currentTriplet: _currentTriplet,
+      currentYear: _currentYear,
+    ));
+  }
+
+  void _onShowFollowing(ShowFollowing event, Emitter<ProfileState> emit) async {
+    if (event.toShow) {
+      _followingList = await UserRepository.getFollowingList();
+    }
+    emit(state.copyWith(
+      following: _followingList,
+      showFollowing: event.toShow,
+    ));
+  }
+
+  static Future follow(String userId) async {
+    final statuscode = await UserRepository.followUser(userId);
+
+    if (statuscode != 200) {
+      throw Exception(AppStrings.genericError);
+    }
+  }
+
+  static Future unfollow(String userId) async {
+    final statuscode = await UserRepository.unfollowUser(userId);
+
+    if (statuscode != 200) {
+      throw Exception(AppStrings.genericError);
+    }
+  }
+
+  // Index -> Index%4
+  static List<String> monthTriplet(int index) {
+    switch (index) {
+      case 0:
+        return ['Jan', 'Feb', 'Mar'];
+
+      case 1:
+        return ['Apr', 'May', 'Jun'];
+
+      case 2:
+        return ['Jul', 'Aug', 'Sep'];
+
+      default:
+        return ['Oct', 'Nov', 'Dec'];
+    }
+  }
+
+  // Cell Color for Acceptance Graph
+  Color getCellColor(DateTime date) {
+    final index = _activity[date] ?? 0;
+    if (index < 0) {
+      return Color.fromRGBO(255, 0, 0, math.min(-0.2 * index, 1));
+    } else if (index > 0) {
+      return Color.fromRGBO(0, 255, 0, math.min(0.2 * index, 1));
+    }
```

Comment:
Can't you just use `math.abs` on the index?

Reply:
See the RGB values; they are different.